File: ClimbingStairs/dp.java
public int climbStairs(int n) {
    if (n < 3) return n;
    int sum1 = 1;
    int sum2 = 2;
    for (int i = 2; i < n; i ++) {
        sum2 = sum1 + sum2;
        sum1 = sum2 - sum1;
    }
    return sum2;
}

File: ClimbingStairs/dp.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def climbStairs(self, n):
        cache = {}
        cache[1], cache[2] = 1, 2

        for i in range(3, n + 1):
            cache[i] = cache[i - 1] + cache[i - 2]

        return cache[n]
